apps/site/assets/ts/hooks/useChannel.ts

```diff
@@ -45,6 +46,22 @@ function useChannel<
     };
   }, [channelId]);
 
+  const isVisible = usePageVisibility();
+  useLayoutEffect(() => {
+    if (channelId !== null) {
+      if (!isVisible) {
+        leaveChannel(channelId);
+      } else {
+        joinChannel<OnDataEventType[]>(channelId, dispatch);
+      }
+    }
+    return () => {
```

Comment:
nitpick: could this return occur after `!isVisible` or does useLayoutEffect require a return?

Reply:
like `useEffect`, `useLayoutEffect` calls the (optional!) return function when the component is destroyed. so it's an appropriate place for any sort of cleanup tasks. Some more documentation [here](https://react.dev/learn/synchronizing-with-effects#how-to-write-an-effect).